library/HTMLPurifier/Lexer/DOMLex.php

```diff
@@ -194,7 +194,7 @@ protected function createStartNode($node, &$tokens, $collect, $config)
         } elseif ($node->nodeType === XML_CDATA_SECTION_NODE) {
             // undo libxml's special treatment of <script> and <style> tags
             $last = end($tokens);
-            $data = $node->data;
+            $data = $node->data ?? "";
```

Comment:
Use of null coalesce operator should be reflected in min PHP version required in `composer.json` (e.g.: `"php": ">=7.0"`), otherwise the library will be broken on older installations.